src/sentry/static/sentry/app/views/eventsV2/results.tsx

```diff
@@ -99,9 +106,70 @@ class Results extends React.Component<Props, State> {
     }
   }
 
-  async fetchTotalCount() {
+  canLoadEvents = async () => {
+    const {api, location, organization} = this.props;
+    const {eventView} = this.state;
+    let needConfirmation = false;
+    let confirmedQuery = true;
+    const currentQuery = eventView.getEventsAPIPayload(location);
+    const duration = eventView.getDays();
+
+    if (duration > 30 && currentQuery.project) {
+      let projectLength = currentQuery.project.length;
+
+      if (
+        projectLength === 0 ||
+        (projectLength === 1 && currentQuery.project[0] === '-1')
+      ) {
+        const results = await fetchProjectsCount(api, organization.slug);
+
+        if (projectLength === 0) projectLength = results.myProjects;
+        else projectLength = results.allProjects;
+      }
+
+      if (projectLength > 10) {
+        needConfirmation = true;
+        confirmedQuery = false;
+      }
+    }
+    // Once confirmed, a change of project or datetime will happen before this can set it to false,
+    // this means a query will still happen even if the new conditions need confirmation
+    // using a state callback to return this to false
+    this.setState({needConfirmation, confirmedQuery}, () => {
+      this.setState({confirmedQuery: false});
+    });
+    if (needConfirmation) {
+      this.openConfirm();
+    }
+    this.fetchTotalCount(confirmedQuery);
+  };
+
+  openConfirm = () => {};
+
+  setOpenFunction = ({open}) => {
+    this.openConfirm = open;
+    return null;
+  };
+
+  handleConfirmed = async () => {
+    this.setState({needConfirmation: false, confirmedQuery: true}, () => {
+      this.setState({confirmedQuery: false});
+    });
+    this.fetchTotalCount(true);
```

Comment:
Should you move the check for a delta in `confirmedQuery` into the `componentDidUpdate` conditional to be consistent with the other views instead of calling fetch directly here? Also, if not, you may want to add a comment explaining why `confirmedQuery` is being set to `true` then back to `false` in the callback here as well (to explain that this is how clicking `okay` on the confirm will actually run the query).

Reply:
Dang good call :+1: this is much cleaner, also means that `canLoadEvents` is also only checking that instead of fetching too :+1: